tools/bazar/controllers/ApiController.php

```diff
@@ -173,6 +174,65 @@ public function createSemanticEntry($formId)
         ]);
     }
 
+    /**
+     * @Route("/api/bazar-list-data", methods={"GET"}, options={"acl":{"public"}})
+     */
+    public function getBazarListData()
+    {
+        try {
+            $bazatListService = $this->getService(BazarListService::class);
+            $forms = $bazatListService->getForms($_GET);
+            $entries = $bazatListService->getEntries($_GET, $forms);     
+            $filters = $bazatListService->formatFilters($_GET, $entries, $forms);  
+            
+            // Basic fields
+            $fieldList = ['id_fiche', 'bf_titre'];
+            // If no id, we need idtypeannonce (== formId) to filter
+            if (!isset($_GET['id'])) $fieldList[] = ['id_typeannonce'];
+            // fields for colo / icon
+            $fieldList = array_merge($fieldList, [ $_GET['colorfield'], $_GET['iconfield'] ]);
+            // Fields for filters
+            foreach($filters as $field => $config) $fieldList[] = $field;
+            // Fields used to search
+            foreach($_GET['searchfields'] ?? [] as $field) $fieldList[] = $field;
+            // Fields used by template
+            foreach($_GET['displayfields'] ?? [] as $field) $fieldList[] = $field;
+            // extra fields required by template
+            $fieldList = array_merge($fieldList, $_GET['necessary_fields'] ?? []);
+            $fieldList = array_values(array_unique(array_filter($fieldList))); // array_values to have incremental keys
+            
+            // Reduce the size of the data sent by transforming entries object into array
+            // we use the $fieldMapping to transform back the data when receiving data in the front end
+            $entries = array_map(function($entry) use ($fieldList) {
+                $result = [];
+                foreach($fieldList as $field) {
+                    $result[] = $entry[$field] ?? null;
+                }
+                return $result;
+            }, $entries);
+        } catch (\Exception $e) {
+            return new ApiResponse($e->getMessage(), Response::HTTP_INTERNAL_SERVER_ERROR);
+        }
+
+        return new ApiResponse(
+            [
+                'entries' => $entries,
+                'fieldMapping' => $fieldList,
+                'filters' => $filters
+            ],
+            Response::HTTP_OK
+        );
+    }
+
+    /**
+     * @Route("/api/entry/{id}/view", methods={"GET"}, options={"acl":{"public"}})
```

Comment:
```suggestion
     * @Route("/api/entry/{id}/view", methods={"GET"}, options={"acl":{"public"}})
```

je suis plutôt opposé à la création de cette route alors qu'il existe déjà une route pour faire la même chose `api/entries/html/{id}` avec `$_GET['fields']='html_output'

Reply:
pour garder les performances, on pourrait modifier `public function getAllFormEntries($formId, $output = null, $selectedEntries = null)
{` en 
```
public function getAllFormEntries($formId, $output = null, $selectedEntries = null)
{
  if (empty($formId) && $output == 'html' && isset($_GET['fields']) && $_GET['fields'] == 'html_output' && !empty($selectedEntries) && count(explode(',',$selectedEntries)) == 1){
        $html = $this->getService(EntryController::class)->view(explode(',',$selectedEntries)[0], '', 1);
        return new ApiResponse(['html_output' => $html]);
}
```

Reply:
J'y repense, il faudrait aussi ajouter un test du genre `$this->getService(AclService::class)->hasAccess('read',$entryId)` pour la route api pour accéder au rendu des fiches